src/components/services/LoanManagerService.ts

```diff
@@ -64,88 +69,158 @@ export const LoanManagerService = (props: Props) => {
-            const token = tokenId != null ? tokens[tokenId] : undefined
-            const denom = token != null ? token.denominations[0] : borrowEngine.currencyWallet.currencyInfo.denominations[0]
-            return div(nativeAmount, denom.multiplier, DECIMAL_PRECISION)
-          }
-          const getCurrencyCode = (tokenId?: string): string => {
-            const tokens = allTokens[borrowEngine.currencyWallet.currencyInfo.pluginId]
-            const token = tokenId != null ? tokens[tokenId] : undefined
-            return token?.currencyCode ?? borrowEngine.currencyWallet.currencyInfo.currencyCode
+  const getCurrencyCode = React.useCallback(
+    (borrowEngine: BorrowEngine, tokenId?: string): string => {
+      const tokens = allTokens[borrowEngine.currencyWallet.currencyInfo.pluginId]
+      const token = tokenId != null ? tokens[tokenId] : undefined
+      return token?.currencyCode ?? borrowEngine.currencyWallet.currencyInfo.currencyCode
+    },
+    [allTokens]
+  )
+
+  // Cache changes to specific watched properties of the loan accounts to detect
+  // deltas
+  useAsyncEffect(async () => {
+    for (const loanAccountId of Object.keys(loanAccountMap)) {
+      await waitForBorrowEngineSync(loanAccountMap[loanAccountId].borrowEngine)
+      const { borrowEngine } = loanAccountMap[loanAccountId]
+      const { debts, collaterals } = borrowEngine
+
+      // Cache accounts only if we can support liquidation price calculations
+      const filteredDebts = debts.filter(debt => !zeroString(debt.nativeAmount))
+      const filteredCollaterals = collaterals.filter(collateral => !zeroString(collateral.nativeAmount))
+      const onlyOneCollateral = filteredCollaterals.length === 1
+      // TODO: Find a less crude way to determine if a token is USD-based
+      const onlyUsdBasedDebts = filteredDebts.every(debt => (debt.tokenId != null ? USD_BASED_TOKEN_IDS.includes(debt.tokenId.toLowerCase()) : false))
+
+      if (onlyOneCollateral && onlyUsdBasedDebts) {
+        // Only trigger push events after the exchange rates are available for
+        // all loan assets on this account
+        const loanAssetTokenIds = [...filteredDebts, ...filteredCollaterals].map(loanAsset => loanAsset.tokenId)
+        if (loanAssetTokenIds.every(tokenId => tokenId == null || !zeroString(exchangeRates[`${getCurrencyCode(borrowEngine, tokenId)}_iso:USD`]))) {
+          if (debts.length > 0) {
+            // If it's the first time the account has been seen, we want to
+            // avoid a notification if it already exceeds the liquidation price
+            // right when the app is booted. Newly created loans can never
+            // exceed the liquidation price.
+            if (Object.keys(cachedLoanAssetsMap).find(accountAssetId => loanAccountId === accountAssetId) == null) {
+              cachedLoanAssetsMap[loanAccountId] = JSON.stringify([...debts, ...collaterals])
+              await uploadLiquidationPushEvents(loanAccountId, false)
```

Comment:
Do this here instead and get rid of the need for the `isSkipPriceCheck` flag param:
```ts
      // TODO: Implement a true way to clear the push event
      if (cachedLoanAssetsMap[loanAccountId] === '') {
        uploadLiquidationEvent('WBTC', 0)
        return
      }
```

Reply:
Also, there is a way to remove the price-change notification, so we should do that as instead of this hack. Let's discuss how.

Reply:
After speaking about the different cases, I'll retract this one because there is a lot of coupling of cases in that function.

Reply:
Because this code may eventually be replaced with server-side handling, I'm not going to push hard on a refactor now.